ModelBundle/Form/DataTransformer/ConditionFromBooleanToMongoTransformer.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace OpenOrchestra\ModelBundle\Form\DataTransformer;
+
+use OpenOrchestra\ModelInterface\Form\DataTransformer\ConditionFromBooleanToBddTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * Class ConditionFromBooleanToMongoTransformer
+ */
+class ConditionFromBooleanToMongoTransformer implements ConditionFromBooleanToBddTransformerInterface
+{
+    protected $field = '';
+
+    /**
+     * @param string $field
+     */
+    public function setField($field)
+    {
+        $this->field = $field;
+    }
+
+    /**
+     * @param string $value
+     *
+     * @return string
+     */
+    public function transform($value)
+    {
+        return $this->transformField(json_decode($value, true));
+    }
+
+    /**
+     * @param string $value
+     *
+     * @return string
+     */
+    public function reverseTransform($value)
+    {
+        return $this->reverseTransformField($value);
+    }
+
+    /**
+     * @param string $conditions
+     *
+     * @return array|string
+     */
+    protected function transformField($conditions)
+    {
+        if (!is_null($conditions)) {
+            foreach($conditions as $key => $condition) {
+                if(is_array($condition)){
+                    $conditions[$key] = $this->transformField($condition);
+                }
+                if (is_null($conditions[$key])) {
+                    return null;
+                }
+                if ($key === '$eq' || $key === $this->field) {
+                    $conditions = $conditions[$key];
+                    break;
+                } elseif ($key === '$ne') {
+                    $conditions = 'NOT '.$conditions[$key];
+                    break;
+                } elseif ($key === '$and') {
+                    $conditions = '( '.implode(' AND ', $conditions[$key]).' )';
+                    break;
+                } elseif ($key === '$or') {
+                    $conditions = '( '.implode(' OR ', $conditions[$key]).' )';
+                    break;
+                }
+            }
+        }
+
+        return $conditions;
+    }
+
+    /**
+     * @param string $condition
+     * @param int    $count
+     * @param array  $aliases
+     * @param string $delimiter
+     *
+     * @return string|null
+     */
+    protected function reverseTransformField($condition, $count = 0, $aliases = array(), $delimiter = '##')
+    {
+        if (!is_null($condition)) {
+            $result = array();
+            $encapsuledElements = array();
+            preg_match_all(ConditionFromBooleanToBddTransformerInterface::GET_BALANCED_BRACKETS, $condition, $encapsuledElements);
+            foreach ($encapsuledElements[0] as $key => $encapsuledElement) {
```

Comment:
@alavieille , preg_match instantiate at last an empty array for index 0 if regular expression not matched